### ISSUE TITLE
Invalid template file helper/summary_short.phtml in module Yotpo_Yotpo

### ISSUE THREAD

**Issue by peterjaap** [YotpoLtd/magento2-plugin#81]
We got this error;

```
Exception #0 (Magento\Framework\Exception\ValidatorException): Invalid template file: 'helper/summary_short.phtml' in module: 'Yotpo_Yotpo' block's name: ''


Exception #0 (Magento\Framework\Exception\ValidatorException): Invalid template file: 'helper/summary_short.phtml' in module: 'Yotpo_Yotpo' block's name: ''
#0 /data/client/magento2/generated/code/Yotpo/Yotpo/Block/Product/YotpoReviewRenderer/Interceptor.php(180): Magento\Framework\View\Element\Template->fetchView(false)
#1 /data/client/magento2/vendor/magento/framework/View/Element/Template.php(300): Yotpo\Yotpo\Block\Product\YotpoReviewRenderer\Interceptor->fetchView(false)
#2 /data/client/magento2/vendor/magento/framework/View/Element/AbstractBlock.php(667): Magento\Framework\View\Element\Template->_toHtml()
#3 /data/client/magento2/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\Element\AbstractBlock->toHtml()
#4 /data/client/magento2/vendor/magento/framework/Interception/Interceptor.php(138): Yotpo\Yotpo\Block\Product\YotpoReviewRenderer\Interceptor->___callParent('toHtml', Array)
#5 /data/client/magento2/vendor/magento/framework/Interception/Interceptor.php(153): Yotpo\Yotpo\Block\Product\YotpoReviewRenderer\Interceptor->Magento\Framework\Interception\{closure}()
#6 /data/client/magento2/generated/code/Yotpo/Yotpo/Block/Product/YotpoReviewRenderer/Interceptor.php(507): Yotpo\Yotpo\Block\Product\YotpoReviewRenderer\Interceptor->___callPlugins('toHtml', Array, Array)
#7 /data/client/magento2/vendor/magento/module-review/Block/Product/ReviewRenderer.php(77): Yotpo\Yotpo\Block\Product\YotpoReviewRenderer\Interceptor->toHtml()
#8 /data/client/magento2/vendor/yotpo/module-review/Block/Product/YotpoReviewRenderer.php(21): Magento\Review\Block\Product\ReviewRenderer->getReviewsSummaryHtml(Object(Magento\Catalog\Model\Product\Interceptor), 'short', true)
#9 /data/client/magento2/generated/code/Yotpo/Yotpo/Block/Product/YotpoReviewRenderer/Interceptor.php(24): Yotpo\Yotpo\Block\Product\YotpoReviewRenderer->getReviewsSummaryHtml(Object(Magento\Catalog\Model\Product\Interceptor), 'short', true)
#10 [internal function]: Yotpo\Yotpo\Block\Product\YotpoReviewRenderer\Interceptor->getReviewsSummaryHtml(Object(Magento\Catalog\Model\Product\Interceptor), 'short', true)
#11 /data/client/magento2/vendor/magento/framework/App/State.php(186): call_user_func_array(Array, Array)
... etc
```

It seems that this line in `vendor/magento/module-review/Block/Product/ReviewRenderer.php` is the culprit;

`$this->setTemplate($this->_availableTemplates[$templateType]);`

This should be;

`$this->setTemplate('Magento_Review::' . $this->_availableTemplates[$templateType]);`

Another option is to change the `$_availableTemplates` var to;

```
    protected $_availableTemplates = [
        self::FULL_VIEW => 'Magento_Review::helper/summary.phtml',
        self::SHORT_VIEW => 'Magento_Review::helper/summary_short.phtml',
    ];
```

Or the template files should be duplicated into the Yotpo extension, which feels dirtier. I'd vote in favor of a plugin that does the first option.

**Comment by peterjaap** [comment]
Another option would be to copy the `getReviewsSummaryHtml` into `YotpoReviewRenderer` and make the edits there.

**Comment by enl** [comment]
I'd rather suggest to fix Magento core issue in this `ReviewRenderer` just because Yotpo is not the only reviews module in the whole world and all of them are affected by this.

As for my project, the issue was introduced only after upgrade to M2.2.

**Comment by chufolopez** [comment]
You must deactivate the module Mage_Review in app/ect/config.php
'Magento_Review' => 0,

**Comment by billykov** [comment]
Hi all,
@chufolopez  does deactivating it solves the issue?

**Comment by chufolopez** [comment]
Yes @billykov .

**Comment by jefflenz** [comment]
That's a bad idea, shouldn't disable core Magento module.  For example, when running php bin/magento module:disable Magento_Review you are met with all the dependency errors.

**Comment by indefinitedevil** [comment]
Does anything in this module work properly? How has this not been fixed yet?

**Comment by stevenkbanks** [comment]
Hello,
I have just found this thread and wondered if anyone could help me with a similar error message.
It appears when we click on Cache Management in Magento 2.2.0.

Very new to all of this so all help very much appreciated!
Thanks, 


1 exception(s):
Exception #0 (Magento\Framework\Exception\ValidatorException): Invalid template file: 'Magento_Backend::system/cache/additional.phtml' in module: 'Magento_Backend' block's name: 'cache.additional'

Exception #0 (Magento\Framework\Exception\ValidatorException): Invalid template file: 'Magento_Backend::system/cache/additional.phtml' in module: 'Magento_Backend' block's name: 'cache.additional'
#0 /home/momentst/public_html/vendor/magento/framework/View/Element/Template.php(300): Magento\Framework\View\Element\Template->fetchView(false)
#1 /home/momentst/public_html/vendor/magento/module-backend/Block/Template.php(124): Magento\Framework\View\Element\Template->_toHtml()
#2 /home/momentst/public_html/vendor/magento/framework/View/Element/AbstractBlock.php(667): Magento\Backend\Block\Template->_toHtml()
#3 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(558): Magento\Framework\View\Element\AbstractBlock->toHtml()
#4 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(534): Magento\Framework\View\Layout->_renderBlock('cache.additiona...')
#5 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(489): Magento\Framework\View\Layout->renderNonCachedElement('cache.additiona...')
#6 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(585): Magento\Framework\View\Layout->renderElement('cache.additiona...')
#7 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(536): Magento\Framework\View\Layout->_renderContainer('content')
#8 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(489): Magento\Framework\View\Layout->renderNonCachedElement('content')
#9 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(585): Magento\Framework\View\Layout->renderElement('content')
#10 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(536): Magento\Framework\View\Layout->_renderContainer('main.col')
#11 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(489): Magento\Framework\View\Layout->renderNonCachedElement('main.col')
#12 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(585): Magento\Framework\View\Layout->renderElement('main.col')
#13 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(536): Magento\Framework\View\Layout->_renderContainer('admin.scope.col...')
#14 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(489): Magento\Framework\View\Layout->renderNonCachedElement('admin.scope.col...')
#15 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(585): Magento\Framework\View\Layout->renderElement('admin.scope.col...')
#16 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(536): Magento\Framework\View\Layout->_renderContainer('page.main.conta...')
#17 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(489): Magento\Framework\View\Layout->renderNonCachedElement('page.main.conta...')
#18 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(585): Magento\Framework\View\Layout->renderElement('page.main.conta...')
#19 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(536): Magento\Framework\View\Layout->_renderContainer('page.content')
#20 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(489): Magento\Framework\View\Layout->renderNonCachedElement('page.content')
#21 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(585): Magento\Framework\View\Layout->renderElement('page.content')
#22 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(536): Magento\Framework\View\Layout->_renderContainer('page.wrapper')
#23 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(489): Magento\Framework\View\Layout->renderNonCachedElement('page.wrapper')
#24 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(585): Magento\Framework\View\Layout->renderElement('page.wrapper')
#25 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(536): Magento\Framework\View\Layout->_renderContainer('backend.page')
#26 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(489): Magento\Framework\View\Layout->renderNonCachedElement('backend.page')
#27 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(585): Magento\Framework\View\Layout->renderElement('backend.page')
#28 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(536): Magento\Framework\View\Layout->_renderContainer('root')
#29 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(489): Magento\Framework\View\Layout->renderNonCachedElement('root')
#30 /home/momentst/public_html/vendor/magento/framework/View/Layout.php(954): Magento\Framework\View\Layout->renderElement('root')
#31 /home/momentst/public_html/generated/code/Magento/Framework/View/Layout/Interceptor.php(37): Magento\Framework\View\Layout->getOutput()
#32 /home/momentst/public_html/vendor/magento/framework/View/Result/Page.php(257): Magento\Framework\View\Layout\Interceptor->getOutput()
#33 /home/momentst/public_html/vendor/magento/framework/View/Result/Layout.php(170): Magento\Framework\View\Result\Page->render(Object(Magento\Framework\App\Response\Http\Interceptor))
#34 /home/momentst/public_html/generated/code/Magento/Backend/Model/View/Result/Page/Interceptor.php(37): Magento\Framework\View\Result\Layout->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#35 /home/momentst/public_html/vendor/magento/framework/App/Http.php(139): Magento\Backend\Model\View\Result\Page\Interceptor->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#36 /home/momentst/public_html/vendor/magento/framework/App/Bootstrap.php(256): Magento\Framework\App\Http->launch()
#37 /home/momentst/public_html/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#38 {main}

**Comment by friendscottn** [comment]
Another temporary fix is to use @peterjaap 's suggestion and write your own module that override's Yotpo's.

app/code/VENDOR/Yotpo/Block/Product/VENDORYotpoReviewRenderer.php
```php
<?php
namespace VENDOR\Yotpo\Block\Product;

// This is a temporary fix for a bad template reference in Yotpo's module
class VENDORYotpoReviewRenderer extends \Yotpo\Yotpo\Block\Product\YotpoReviewRenderer
{
    protected $_availableTemplates = [
        self::FULL_VIEW => 'Magento_Review::helper/yotposummary.phtml',
        self::SHORT_VIEW => 'Magento_Review::helper/summary_short.phtml',
    ];
    
}
```

Then use the app/code/VENDOR/Yotpo/etc/frontend/di.xml file to prefer yours class.
```xml
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
    <preference for="Yotpo\Yotpo\Block\Product\YotpoReviewRenderer" type="VENDOR\Yotpo\Block\Product\VENDORYotpoReviewRenderer" />
</config>
```

**Comment by jefflenz** [comment]
Just create the PHTML files in your theme, no need for any modules or override.

**Comment by jefflenz** [comment]
It seems the developers have abandoned this extension

**Comment by silversil** [comment]
I got the same issue but with this error:

1 exception(s):
Exception #0 (Magento\Framework\Exception\ValidatorException): Invalid template file: 'Magento_Backend::system/cache/additional.phtml' in module: 'Magento_Backend' block's name: 'cache.additional'

Exception #0 (Magento\Framework\Exception\ValidatorException): Invalid template file: 'Magento_Backend::system/cache/additional.phtml' in module: 'Magento_Backend' block's name: 'cache.additional'

HERE MY WORKING SOLUTION:
in production server was missing this folder:
vendor/magento/module-backend/view/adminhtml/templates/system/cache

with 2 files inside: additional.phtml e edit.phtml

I have created the folder and I have copyed the files from developer installation, now it'w working.

Enjoy.

**Comment by JanuszWitrykus** [comment]
This is so simple fix. Why they did not it since today?

**Comment by DanGlz** [comment]
fixed in #106 PR
